code/src/pages/products/ProductsPage.test.tsx

```diff
@@ -0,0 +1,65 @@
+import { MemoryRouter } from "react-router-dom";
+import { render, screen } from "@testing-library/react";
+import ProductsPage from "@/pages/products/ProductsPage";
+import { useGetProductsQuery } from "@/store/api/productsApi";
+
+jest.mock("@/store/api/productsApi", () => ({
+  useGetProductsQuery: jest.fn()
+}));
+
+jest.mock("@/components/product-card/ProductCard", () =>
+  jest.fn(() => <div data-testid="product-card" />)
+);
+
+const mockProducts = [
+  { id: 1, name: "Product 1", price: 100 },
+  { id: 2, name: "Product 2", price: 200 },
+  { id: 3, name: "Product 3", price: 300 },
+  { id: 4, name: "Product 4", price: 400 }
+];
+
+const renderProductsPage = () =>
+  render(
+    <MemoryRouter>
+      <ProductsPage />
+    </MemoryRouter>
+  );
+
+describe("ProductsPage", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test("renders products when data is available", () => {
+    (useGetProductsQuery as jest.Mock).mockReturnValue({
+      data: mockProducts,
+      isLoading: false
+    });
+
+    renderProductsPage();
+
+    expect(screen.getAllByTestId("product-card")).toHaveLength(
+      mockProducts.length
+    );
+  });
+
+  test("renders the product count correctly", () => {
+    renderProductsPage();
+    (useGetProductsQuery as jest.Mock).mockReturnValue({
+      data: mockProducts,
+      isLoading: false
+    });
+
+    expect(screen.getByText(mockProducts.length)).toBeInTheDocument();
+  });
+
+  test("renders the sort by section", () => {
+    renderProductsPage();
+    (useGetProductsQuery as jest.Mock).mockReturnValue({
+      data: mockProducts,
+      isLoading: false
+    });
+
+    expect(screen.getByText(/Sort by: Recommended/i)).toBeInTheDocument();

```

Comment:
> Move into separate variable. Also there is not need to use different casing if you use `i` regex flag, just make all lowercase



> Move into separate variable. Also there is not need to use different casing if you use `i` regex flag, just make all lowercase

moved to variable and I didn't change value it's so not important how I wrote property inside upper or lower case because we have `i` flag